### PR TITLE
Add internal function to rebalancingSetToken that clears state after …

### PR DESCRIPTION
**Pull request by bweick**
…rebalance is completed (failure or success).

### CODE REVIEW

File: contracts/core/tokens/RebalancingSetToken.sol [hunk 1]
```diff
@@ -619,4 +621,24 @@ contract RebalancingSetToken is
...
+        startingCurrentSetAmount = 0;
+        delete auctionParameters;
+
+        if (rebalanceState == RebalancingHelperLibrary.State.Drawdown) {
+            biddingParameters.minimumBid = 0;
+            biddingParameters.remainingCurrentSets = 0;
+            delete biddingParameters.combinedCurrentUnits;
+            delete biddingParameters.combinedNextSetUnits;
+        } else {
+            delete biddingParameters;
```
Comment (felix2feng) [inline contracts/core/tokens/RebalancingSetToken.sol]: What's the difference between the two cases? Don't they do the same thing?

File: contracts/core/tokens/RebalancingSetToken.sol [hunk 2]
```diff
@@ -619,4 +621,24 @@ contract RebalancingSetToken is
     {
         return biddingParameters.combinedNextSetUnits;
     }
+
+    /* ============ Getter Functions ============ */
+
+    function clearAuctionState()
```
Comment (felix2feng) [inline contracts/core/tokens/RebalancingSetToken.sol]: Add java doc

File: contracts/core/tokens/RebalancingSetToken.sol [hunk 3]
```diff
@@ -619,4 +621,24 @@ contract RebalancingSetToken is
     {
         return biddingParameters.combinedNextSetUnits;
     }
+
+    /* ============ Getter Functions ============ */
```
Comment (felix2feng) [inline contracts/core/tokens/RebalancingSetToken.sol]: Fix naming

File: contracts/core/tokens/RebalancingSetToken.sol (hunk 1 above)
Comment (bweick) [reply contracts/core/tokens/RebalancingSetToken.sol]: We need the `combinedTokenArray` for the `withdrawFromFailedRebalance`

Comment (felix2feng) [reply contracts/core/tokens/RebalancingSetToken.sol]: Hmm - why don't we delete `biddingParameters` and generate a variable called `withdrawComponents` or something for withdrawing?

Comment (felix2feng) [reply contracts/core/tokens/RebalancingSetToken.sol]: @asoong What do you think?

Comment (asoong) [reply contracts/core/tokens/RebalancingSetToken.sol]: Yeah not a fan of this, also not a fan of duplicating the state. Ideally we store them separately and only delete `minimumBid` and `remainingCurrentSets` but i'm guessing they're grouped together now so they can be passed around more efficiently

Comment (asoong) [reply contracts/core/tokens/RebalancingSetToken.sol]: Actually, they're only passed together when placing bid and getting bid price

**Pull request merged**